src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -238,22 +254,34 @@ export class EditorComponent implements OnInit {
     //  Only gets the image data from (0,0) to (width,height) of image.
     this.originalImageData = this.maskCtx.getImageData(0, 0, imgWidth, imgHeight);
     this.maskCtx.clearRect(0,0,imgWidth, imgHeight);
-    
+
     this.drawScaledImage();
 
     // If there is a mask URL passed in then draw mask.
     if (this.maskUrl != '' && this.maskUrl) {
-      console.log('there\'s a mask url' + this.maskUrl);
+      console.log("there's a mask url" + this.maskUrl);
       let maskImage = new Image();
       maskImage.onload = () => {
         this.maskCtx.drawImage(maskImage, 0, 0);
-        this.maskImageData = this.maskCtx.getImageData(0, 0, maskImage.width, maskImage.height);
+        this.maskImageData = this.maskCtx.getImageData(
+          0,
+          0,
+          maskImage.width,
+          maskImage.height
+        );
         this.drawMask();
-      }
+      };
       maskImage.src = this.maskUrl;
     }
-  }
 
+    this.allPixels = new Set([
+      ...Array.from(Array(this.originalImageData.data.length / 4).keys()).map(
+        function (x) {
+          return x * 4;
+        }
```

Comment:
It would be useful to add a comment about what this is doing for future reference.